### ISSUE TITLE
Patient register questions and RDT input by CHW viewable in the UI 

### ISSUE THREAD

**Issue by vincent-karuri** [onaio/rdt-standard#173]
SID wants to be able to view the patient register information and the RDT taken for the individual on the UI. This study will be conducted by Ministry of health official health care workers, they may be incorporating it as part of their routine analysis of their patients or request to view the data.

**Comment by vincent-karuri** [comment]
@shivenbhatt1 instead of implementing this for the register screen, would it be easier to ask @rowo to help design a screen that displays all tests taken for the patient.

The tests list can be accessed from the register and the list of rdt tests can have information about rdt id, results, date etc. and a button to add PCR results related to a particular rdt test.

**Comment by shivenbhatt1** [comment]
@vincent-karuri @rowo I vaguely understand what you're saying here, and I'm thinking this would work.  If there is an example of this could you share it?

**Comment by vincent-karuri** [comment]
I don't know if there's an example but it would involve something like navigating to a client's profile page and viewing all tests ever administered to them.

Then each test can have an edit button to add pcr information.

@rowo would know better

**Comment by rowo** [comment]
@vincent-karuri I don't have a lot of context here, but are you talking about something like this? 

![Profile - RDT](https://user-images.githubusercontent.com/1584163/65159483-25c57c80-da02-11e9-935e-4f4f2e736a7b.png)

**Comment by vincent-karuri** [comment]
@rowo yes, this is the idea. Although I don't think we want to conduct an RDT test from this for as well.

CC @shivenbhatt1

**Comment by rowo** [comment]
@vincent-karuri okay, if it doesn't make sense to do RDT that's okay with me. In other OpenSRP it's something we do since the list button is more of a shortcut (kind of like deleting emails from the list or the individual email view) but I can understand how this is different.

**Comment by shivenbhatt1** [comment]
@rowo @vincent-karuri This is exactly what we need.  I'd also say Indonesia will require historical records of Laboratory Data (i.e. Microscopy Results (simple +/- result for now), qPCR (simple +/- result for now), and Sequencing linked to their respective RDT ID).  I'm not sure if you want to go ahead and make a one all, or make the mockup in segments.  So for the patient, we'd have a historical list of RDTs conducted with results, historical list of microscopy associated with that RDT and results, list of qPCR results.  ***I think for now sequencing can be skipped but this would be super cool for the future to know what strain and drug resistance that patient had been carrying directly pushed to the health worker so that they can better administer drugs.

**Comment by rowo** [comment]
@shivenbhatt1 can you write out example content for a typical user including any information that would be displayed such as date or multiple tests? I think we can mockup the one all and also a simple version (like the above) to see if they can be treated consistently.

**Comment by shivenbhatt1** [comment]
@rowo @vincent-karuri 

There may be multiple RDTs for each types for each patient: OG RDT and CareStart.  Each RDT conducted on the patient will have their respective RDT IDs, dates that they were conducted would be good here.

For Each Conducted RDT the will associated values will be:
1.  RDT Reported Result: Positive/Negative/Invalid
2. qPCR (BloodSpot) . Notes here: We will be tracking pos/neg/invalid + and also threshold High/low/none
3. qPCR (Directly from RDT) . Notes here: We will be tracking pos/neg/invalid + and also threshold High/low/none
4. Microscopy Result: Pos/Neg/Invalid

In Summary, once the user clicks on patient name, they will either have the option of conducting a new RDT, or viewing previous tests.  From the Previous test section, everything will be linked through the RDT ID and and organized by date, maybe also include whether it is OG RDT or CareStart.  When RDT IDs are selected, the user will see the associated values above 1-4.

**Comment by shivenbhatt1** [comment]
Microscopy comes in counts per white blood cell value, Isabella will advice on how this looks.  We can set pos/neg/invalid and they set a numeric value of counts of malaria/blood

**Comment by rowo** [comment]
@shivenbhatt1 can you actually just write out example content that shows the various and extremes of content possibilities in each row?

**Comment by vincent-karuri** [comment]
https://marvelapp.com/project/4542567/

**Comment by vincent-karuri** [comment]
@rowo in the designs above, how do we access the test result screen from the register page?

**Comment by rowo** [comment]
@vincent-karuri I added in the [register list page](https://marvelapp.com/5e396d7/screen/65213639) so it's clearer. The test results for each RDT show up on the [profile](https://marvelapp.com/5e396d7/screen/63867479), then the user can tap into them to see [details](https://marvelapp.com/5e396d7/screen/65213646).

**Comment by vincent-karuri** [comment]
Hmm, ok. So now the RDT button does not directly launch the tests but launches this [page](https://marvelapp.com/5e396d7/screen/63867479) from which you can then launch the normal test flow?

**Comment by rowo** [comment]
@vincent-karuri The RDT button does launch the tests. Tapping the row goes to the [profile](https://marvelapp.com/5e396d7/screen/63867479), but tapping the button goes to the RDT test flow.

**Comment by vincent-karuri** [comment]
Ah ok, thanks

**Comment by vincent-karuri** [comment]
@shivenbhatt1 for this [page](https://marvelapp.com/5e396d7/screen/65213646), do we only show qpcr information for the openrdt?

I see that in this case we have both a carestart rdt result and an open guideline qpcr result?

**Comment by shivenbhatt1** [comment]
This page should technically read CareStart qPCR.  They are respective to the test, CareStart RDT = CareStart qPCR, OG RDT = OG RDT qPCR

**Comment by vincent-karuri** [comment]
@shivenbhatt1 on this [page](https://marvelapp.com/5e396d7/screen/65213646) can we have just `pv positive` and assume absence of a result means negative?

Also, is microscopy just a `positive/negative` value and can never hold other values?

**Comment by shivenbhatt1** [comment]
@vincent-karuri They can, has to mimic the qPCR list on the patient view. They have different stains to show the different species of malaria as suggested by their microscopy sheet https://docs.google.com/spreadsheets/d/1oMcTIe_mqcpm8ibJjlUfCEOQfpe-OG76/edit#gid=226928130

**Comment by shivenbhatt1** [comment]
@vincent-karuri Add the last column as "Pf.g" to resemble "Pf. gameto" onto the list observed in the patient view as well
